frame/ibc/ibc-rpc/src/lib.rs

```diff
@@ -1234,4 +1238,41 @@ where
 	) -> Result<QueryDenomTracesResponse> {
 		Err(runtime_error_into_rpc_error("Unimplemented"))
 	}
+
+	fn query_newly_created_clients(
+		&self,
+		block_hash: Block::Hash,
+	) -> Result<Vec<IdentifiedClientState>> {
+		let api = self.client.runtime_api();
+		let at = BlockId::Hash(block_hash);
+		let events = api
+			.block_events(&at)
+			.map_err(|_| runtime_error_into_rpc_error("[ibc_rpc]: failed to read block events"))?;
+
+		let mut identified_clients = vec![];
+		for e in events {
+			match e {
+				pallet_ibc::events::IbcEvent::CreateClient { client_id, .. } => {
+					let result: ibc_primitives::QueryClientStateResponse = api
+						.client_state(&at, client_id.clone())
+						.ok()
```

Comment:
do you need this `ok`? there's a flatten, and then an `ok_or_else` afterwards

Reply:
That `client_state` call returns a`Result<Option>`, I call  `ok()` to turn it into an `Option<Option<_>>` , so I can flatten it and handle an error only once.